src/client/index.ts

```diff
@@ -350,11 +364,11 @@ export function getLocationBasedUri(
   return `${location}${property}`;
 }
 
-export function getTokenUri(clientProps: ClientProps): string {
-  if (clientProps.tokenExchangePath) {
-    return `${clientProps.url}${clientProps.tokenExchangePath}`;
+export function getTokenUri(clientConfig: ClientConfig): string {
+  if (clientConfig.tokenExchangePath) {
+    return `${clientConfig.url}${clientConfig.tokenExchangePath}`;
   }
-  return `${clientProps.url}/realms/${clientProps.realm}/protocol/openid-connect/token`;
+  return `${clientConfig.url}/realms/${clientConfig.realm}/protocol/openid-connect/token`;
 }
```

Comment:
What URI is this function providing actually? Looking at `.env`, looks like the `tokenExchangePath` configuration is the Tunnistamo specific (?) `/api-tokens` endpoint. If `tokenExchangePath` is falsy (the `if` block is skipped), then the returned URI looks more like a (Keycloak) OpenID Connect Token endpoint. But AFAIK, those aren't the same thing. This confuses me.

Reply:
This is not a new function- just props renamed.

It provides an URI to get tokens. :D If realm is refined, the server is considered to be  a Keycloak server, which can provide tokens too, but not to be used like apiAccessTokens at the moment.

Reply:
Perhaps it's ok to leave this as is, because as you said, this function isn't introduced (nor logic changed) in this PR. But it still confuses me.

The word "token" is probably too overloaded. There are tokens all over the place in OIDC. OIDC has a standard token exchange endpoint defined by [RFC8693](https://datatracker.ietf.org/doc/html/rfc8693). Both Tunnistamo and Keycloak implement this service. Tunnistamo additionally has a non-standard api-tokens endpoint (and if I'm not mistaken, Keycloak also has a non-standard endpoint for something similar, but off course it's different from the Tunnistamo version). So, I'm confused what tokens this `getTokenUri` is really dealing with. Or then I'm completely lost with all these token standards and non-standards, that's possible too.

Reply:
The amount of tokens confuses me too, because we have endpoints serving standard and non-standard tokens and all are named "***token". I'll leave this code as is, because it is returning correct token path for server with realms (Keycloak) and server with given token path.